### ISSUE TITLE
`SetPartitions()` should not error out in degenerate cases

### ISSUE THREAD

**Issue by maxale** [sagemath/sage#37643]
### Steps To Reproduce

In contrast to `Partitions()` and `OrderedSetPartitions`, `SetPartitions()` imposes unnecessary restrictions on arguments and errors out when they are not satisfied:
```
sage: list( Partitions(3,length=4) )
[]
sage: list( OrderedSetPartitions(range(3),4) )
[]
sage: list( SetPartitions(range(3),4) )
...
ValueError: part must be <= len(set)
```

### Expected Behavior

In the above example, `SetPartitions(range(3),4)` should simply return an empty object (as there are no suitable set partitions). It is a degenerate case, but not an error.

### Actual Behavior

`SetPartitions(range(3),4)` errors out.

### Environment

```markdown
- **OS**: Ubuntu 22.04.4 LTS
- **Sage Version**: 10.3
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by DitoluT** [comment]
This problem can actually be solved by returning an empty setPartition, if length is greater it means there exists no partitions

**Comment by Nolord** [comment]
On could remove the `if` condition that raises [here](https://github.com/sagemath/sage/blob/develop/src/sage/combinat/set_partition.py#L2067) and place it [here](https://github.com/sagemath/sage/blob/develop/src/sage/combinat/set_partition_iterator.pyx#L132) (at the level of the iterator in the Cython code) as such
```Python
if n >= k: #here
   for P in _set_partition_block_gen(n, k, a): #additional indent
      yield from_word(<list> P, base) #additional indent
```

**Comment by S17A05** [comment]
Fixed in 10.4beta4 via #37781.
